src/coreclr/jit/gentree.cpp

```diff
@@ -23591,6 +23591,19 @@ GenTree* Compiler::gtNewSimdShuffleNode(
             // swap the operands to match the encoding requirements
             retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512BW_PermuteVar32x16, simdBaseJitType, simdSize);
         }
+        else if (elementSize == 1)
+        {
+            for (uint32_t i = 0; i < elementCount; i++)
+            {
+                vecCns.u8[i] = (uint8_t)(vecCns.u8[i * elementSize] / elementSize);
+            }
+
+            op2                        = gtNewVconNode(type);
+            op2->AsVecCon()->gtSimdVal = vecCns;
+
+            // swap the operands to match the encoding requirements
+            retNode = gtNewSimdHWIntrinsicNode(type, op2, op1, NI_AVX512VBMI_PermuteVar64x8, simdBaseJitType, simdSize);
+        }
```

Comment:
It would be nice to extract this acceleration into its own PR and also cover Vector256 at the same time, if possible.

Reply:
Moved to separate PR